docs/developers/docs/upgrading.md

```diff
@@ -17,6 +17,97 @@ on our Github repo.
 
 Your current version can be found in `Pupilfirst::Application::VERSION`.
 
+### 2021.4
+
+Introduced support for public and private file storage. Its now mandatory to have two bucket's for storage, one public and one private linked to the app.
+
+#### Migration Steps
+
+<!-- Improve documentation -->
+
+1. Convert the current bucket that you are using to a public bucket and create a new private bucket.
+2. Rename the env variable `AWS_BUCKET` to `AWS_BUCKET_PUBLIC`.
+3. Add a new environment variable `AWS_BUCKET_PRIVATE`, this should be the private bucket name. (New Bucket)
+4. Deploy and run the migration.
+
+#### Steps for configuring public access settings for your S3 bucket
+
+Amazon S3 Block Public Access feature provides settings to help you manage public access to Amazon S3 resources. By default, new buckets and objects do not allow public access.
+You can use the S3 console or AWS CLI to configure public access settings for your bucket. 
+
+For information on configuring public access to your S3 storage, please refer to the documentation [here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/configuring-block-public-access-bucket.html)
+
+#### Creating S3 private bucket
+When you create a new bucket through the AWS S3 console, the default setting in set permissions is Block all public access.
```

Comment:
@prashanth726 : As _Block all public access_ is a setting that is being referred to , it's better to either quote or italicize it.